railties/test/application/test_runner_test.rb

```diff
@@ -14,7 +15,7 @@ def teardown
       teardown_app
     end
 
-    def test_run_in_test_environment
+    def test_run_in_differing_environments
```

Comment:
@kaspth do you think it's worth keep the other test around to be sure it runs in test env by default?